addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -677,6 +679,85 @@ private boolean performImageTagAttack(HtmlContext context, HttpMessage msg, Stri
         return false;
     }
 
+    private boolean performAttributeAttack(HtmlContext context, HttpMessage msg, String param) {
+        String attackString = ACCESSKEY_ATTRIBUTE_ALERT;
+        List<HtmlContext> context2 = performAttack(msg, param, attackString, context, 0);
+        if (context2 == null) {
+            return false;
+        }
+        for (HtmlContext ctx : context2) {
+            Map<String, String> attrMap = ctx.getTagAttributes();
+            if (attrMap.containsKey("accesskey")
+                    && attrMap.get("accesskey").equals("x")
+                    && attrMap.containsKey("onclick")
+                    && attrMap.get("onclick").equals("alert(1)")
+                    && attrMap.containsKey("b")
+                    && attrMap.get("b").equals("")) {
+                // Yep, its vulnerable
+                newAlert()
+                        .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                        .setParam(param)
+                        .setAttack(ctx.getTarget())
+                        .setEvidence(ctx.getTarget())
+                        .setMessage(ctx.getMsg())
+                        .setOtherInfo(
+                                Constant.messages.getString(MESSAGE_PREFIX + "otherinfo.accesskey"))
+                        .raise();
```

Comment:
Worth extracting a method.

Reply:
Other tests don't have it as a method :). Should I change it for them too?

Reply:
Like a setAlertMethod() will be good actually :). I can create that with all the params necessary :)

Reply:
Just change the code being added, to keep the PR focused.

Reply:
I was suggesting, e.g. `raiseAccessKeyAlert(String param, HtmlContext ctx)` to avoid duplicating exactly the same code, i.e.:
```java
newAlert()
    .setConfidence(Alert.CONFIDENCE_MEDIUM)
    .setParam(param)
    .setAttack(ctx.getTarget())
    .setEvidence(ctx.getTarget())
    .setMessage(ctx.getMsg())
    .setOtherInfo(Constant.messages.getString(MESSAGE_PREFIX + "otherinfo.accesskey"))
    .raise();
```

See following review comments.

Reply:
I have changed it to raiseAlert() with the following:
```
    private void raiseAlert(int confidence, String param, HtmlContext ctx, String otherInfo) {
        newAlert()
                .setConfidence(confidence)
                .setParam(param)
                .setAttack(ctx.getTarget())
                .setEvidence(ctx.getTarget())
                .setMessage(ctx.getMsg())
                .setOtherInfo(otherInfo)
                .raise();
    }

```

Reply:
It can be used more generally :)
and can also lead to some GFIs :)

Reply:
Would also make the code look less scary :P

Reply:
Isn't confidence always the same? (In which case it doesn't need to be passed)

Reply:
Uhh for this case its same

Reply:
For some cases its HIGH

Reply:
Low cases are also there :)

Reply:
GG then? Or should I change it :)?

Reply:
Please add `raiseAccessKeyAlert` to avoid duplicating the other info.

Reply:
I meant to add a new method not replace the `raiseAlert`.

Reply:
Uhhhh :)

Reply:
Ouch :)

Reply:
added it back

Reply:
Now gg? :)

Reply:
lgtm